examples/overview_of_the_different_samplers.ipynb

```diff
@@ -39,7 +52,7 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 2,
+   "execution_count": 108,
```

Comment:
(very picky) Could you fully execute the notebook from scratch so to make the execution count reasonable?

Reply:
> LGTM, this is a useful feature.
> 
> I wonder whether we could develop a generalized sampler based on machine learning surrogates (i.e. regressors). Something like:
> 
> ```python
> class Regressor(ABC):
> 
>     @abstractmethod
>     def fit(self, X, y):
>         raise NotImplementedError
> 
>     @abstractmethod
>     def predict(x):
>         raise NotImplementedError
> 
> 
> class SurrogateSampler(BaseSampler):
> 
> 
>     def __init__(  # pylint: disable=too-many-arguments
>         self,
>         batch_size: int,
>         regressor: Regressor,
>         random_state: Optional[int] = None,
>         max_deduplication_passes: int = 5,
>         candidate_pool_size: Optional[int] = None):
>         ...
>         self.regressor = regressor
> ```

Yes I completely agree, it's something that I also thought I would do, but I wanted to do this in a different PR